python/tvm/relax/op/base.py

```diff
@@ -344,3 +344,20 @@ def shape_of(expr: Expr) -> Expr:
         A relax Call, which gets the shape of the input
     """
     return _ffi_api.shape_of(expr)  # type: ignore # pylint: disable=no-member
+
+
+def register_gradient(op_name, fgradient=None, level=10):
+    """Register operator gradient function for a relax operator.
+
+    Parameters
+    ----------
+    op_name : str
+        The name of the op.
+
+    fgradient : function (orig_expr : Expr, output_grad : Expr) -> new_expr : Expr
+        The gradient being used.
+
+    level : int
+        The priority level
```

Comment:
I have little knowledge on AD, so am curious about the meaning of the priority level here.

Reply:
Actually this is inherited from `relay.op.register_gradient` :)

AFAIK, this is required when registering op attrs. When `level=0`, this attribute is not used. We can register the same type of attribute with higher priority level for the same op, so the old attribute would be overwritten.

Reply:
In my view `plevel` is not for AD specially. In tvm when we register or update attributes, we must specify `attr_name`, `key`, `value` and `plevel`. Since fgradient is just registered as an attribute of op (attr_key=`FPrimalGradient`), it also has `plevel`.